ray-operator/controllers/ray/rayjob_controller_test.go

```diff
@@ -33,45 +33,112 @@ import (
-	ctx := context.TODO()
-	myRayJob := &rayiov1alpha1.RayJob{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      "rayjob-test",
-			Namespace: "default",
-		},
-		Spec: rayiov1alpha1.RayJobSpec{
-			Entrypoint: "sleep 999",
-			RayClusterSpec: &rayiov1alpha1.RayClusterSpec{
-				RayVersion: "1.12.1",
-				HeadGroupSpec: rayiov1alpha1.HeadGroupSpec{
-					Replicas: pointer.Int32(1),
+var myRayJob = &rayiov1alpha1.RayJob{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      "rayjob-test",
+		Namespace: "default",
+	},
+	Spec: rayiov1alpha1.RayJobSpec{
+		Entrypoint: "sleep 999",
+		RayClusterSpec: &rayiov1alpha1.RayClusterSpec{
+			RayVersion: "1.12.1",
+			HeadGroupSpec: rayiov1alpha1.HeadGroupSpec{
+				Replicas: pointer.Int32(1),
+				RayStartParams: map[string]string{
+					"port":                        "6379",
+					"object-store-memory":         "100000000",
+					"dashboard-host":              "0.0.0.0",
+					"num-cpus":                    "1",
+					"node-ip-address":             "127.0.0.1",
+					"dashboard-agent-listen-port": "52365",
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{
+							"groupName": "headgroup",
+						},
+						Annotations: map[string]string{
+							"key": "value",
+						},
+					},
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{
+							{
+								Name:  "ray-head",
+								Image: "rayproject/ray:2.4.0",
+								Env: []corev1.EnvVar{
+									{
+										Name: "MY_POD_IP",
+										ValueFrom: &corev1.EnvVarSource{
+											FieldRef: &corev1.ObjectFieldSelector{
+												FieldPath: "status.podIP",
+											},
+										},
+									},
+								},
+								Resources: corev1.ResourceRequirements{
+									Limits: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("1"),
+										corev1.ResourceMemory: resource.MustParse("2Gi"),
+									},
+									Requests: corev1.ResourceList{
+										corev1.ResourceCPU:    resource.MustParse("1"),
+										corev1.ResourceMemory: resource.MustParse("2Gi"),
+									},
+								},
+								Ports: []corev1.ContainerPort{
+									{
+										Name:          "gcs-server",
+										ContainerPort: 6379,
+									},
+									{
+										Name:          "dashboard",
+										ContainerPort: 8265,
+									},
+									{
+										Name:          "head",
+										ContainerPort: 10001,
+									},
+									{
+										Name:          "dashboard-agent",
+										ContainerPort: 52365,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			WorkerGroupSpecs: []rayiov1alpha1.WorkerGroupSpec{
+				{
+					Replicas:    pointer.Int32(3),
+					MinReplicas: pointer.Int32(0),
+					MaxReplicas: pointer.Int32(10000),
+					GroupName:   "small-group",
 					RayStartParams: map[string]string{
 						"port":                        "6379",
-						"object-store-memory":         "100000000",
```

Comment:
Again, there is no change to the spec actually. This holds true for all subsequent code as well.